src/lib/holocene/markdown-editor/markdown-editor.svelte

```diff
@@ -0,0 +1,80 @@
+<script lang="ts">
+  import { markdown } from '@codemirror/lang-markdown';
+  import { EditorState } from '@codemirror/state';
+  import { EditorView, lineNumbers } from '@codemirror/view';
+  import { onMount, tick } from 'svelte';
+
+  import TabButton from '$lib/holocene/tab-buttons/tab-button.svelte';
+  import TabButtons from '$lib/holocene/tab-buttons/tab-buttons.svelte';
+  import { useDarkMode } from '$lib/utilities/dark-mode';
+  import { getEditorThemeWithLineNumbers } from '$lib/vendor/codemirror/custom-extensions';
+
+  import MarkdownPreview from './markdown-preview.svelte';
+
+  interface Props {
+    content: string;
+  }
+
+  let { content = $bindable('') }: Props = $props();
+  let editorElement: HTMLElement | null = $state(null);
+  let editorView: EditorView | null = $state(null);
+  let activeTab: 'edit' | 'preview' = $state('edit');
+
+  const createEditorView = () => {
+    return new EditorView({
+      parent: editorElement,
```

Comment:
You could make this function pure if you passed editorElement in instead of depending on the state being declared already.